File: backend/kanban-board-task-management/TaskManagementSystem/src/main/java/com/example/kanban/service/AdminService.java
package com.example.kanban.service;

import org.springframework.data.domain.Page;

import com.example.kanban.dto.admin.AdminDashboardResponseDto;
import com.example.kanban.dto.task.TaskResponseDto;
import com.example.kanban.dto.user.UserResponseDto;

public interface AdminService {

	// User Management
	Page<UserResponseDto> getAllUsers(int page, int size, String sortBy, String direction);

	UserResponseDto getUserById(Long userId);

	void deleteUser(Long userId);

	// Task Management
	Page<TaskResponseDto> getAllTasks(int page, int size, String sortBy, String direction);

	TaskResponseDto getTaskById(Long taskId);

	void deleteTask(Long taskId);
	
	//Dash board
	AdminDashboardResponseDto getDashboard();

}

File: frontend/kanban-board-task-management/src/services/adminService.js
import axiosInstance from "../api/axiosConfig";

export const getAdminDashboard = async () => {
  const response = await axiosInstance.get("/api/admin/dashboard");
  return response.data;
};

export const getAllUsers = async (
  page = 0,
  size = 5,
  sortBy = "createdAt",
  direction = "desc",
) => {
  const response = await axiosInstance.get("api/admin/users", {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  });
  return response.data;
};
export const getUserById = async (userId) => {
  const response = await axiosInstance.get(`/api/admin/users/${userId}`);
  return response.data;
};
export const deleteUser = async (userId) => {
  const response = await axiosInstance.delete(`/api/admin/users/${userId}`);
  return response.data;
};
export const getAllTasks = async (
  page = 0,
  size = 5,
  sortBy = "createdAt",
  direction = "desc",
) => {
  const response = await axiosInstance.get("/api/admin/tasks", {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  });

  return response.data;
};
export const getTaskById = async (taskId) => {
  const response = await axiosInstance.get(`/api/admin/tasks/${taskId}`);
  return response.data;
};
export const deleteTask = async (taskId) => {
  const response = await axiosInstance.delete(`/api/admin/tasks/${taskId}`);
  return response.data;
};
